File: services/attendance-service/src/test/java/com/visor/school/attendance/model/AttendanceStatusTest.java
package com.visor.school.attendance.model;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

import java.util.Arrays;

class AttendanceStatusTest {

    @Test
    void shouldHaveAllRequiredStatusValues() {
        AttendanceStatus[] statuses = AttendanceStatus.values();

        assertTrue(Arrays.asList(statuses).contains(AttendanceStatus.PRESENT));
        assertTrue(Arrays.asList(statuses).contains(AttendanceStatus.ABSENT));
        assertTrue(Arrays.asList(statuses).contains(AttendanceStatus.LATE));
        assertTrue(Arrays.asList(statuses).contains(AttendanceStatus.EXCUSED));
        assertEquals(4, statuses.length);
    }

    @Test
    void shouldHaveCorrectStatusNames() {
        assertEquals("PRESENT", AttendanceStatus.PRESENT.name());
        assertEquals("ABSENT", AttendanceStatus.ABSENT.name());
        assertEquals("LATE", AttendanceStatus.LATE.name());
        assertEquals("EXCUSED", AttendanceStatus.EXCUSED.name());
    }
}

File: services/attendance-service/src/test/kotlin/com/visor/school/attendanceservice/model/AttendanceStatusTest.kt
package com.visor.school.attendanceservice.model

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class AttendanceStatusTest {

    @Test
    fun `should have all required status values`() {
        val statuses = AttendanceStatus.values()

        assertTrue(statuses.contains(AttendanceStatus.PRESENT))
        assertTrue(statuses.contains(AttendanceStatus.ABSENT))
        assertTrue(statuses.contains(AttendanceStatus.LATE))
        assertTrue(statuses.contains(AttendanceStatus.EXCUSED))
        assertEquals(4, statuses.size)
    }

    @Test
    fun `should have correct status names`() {
        assertEquals("PRESENT", AttendanceStatus.PRESENT.name)
        assertEquals("ABSENT", AttendanceStatus.ABSENT.name)
        assertEquals("LATE", AttendanceStatus.LATE.name)
        assertEquals("EXCUSED", AttendanceStatus.EXCUSED.name)
    }
}

